Docs: Getting started category

nanjekyejoannah opened the pull request:
<!-- Regorganize content for the `Getting Started` category -->

#### Brief description of what is fixed or changed

This PR categorizes the docs according to the diataxis format.

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

Commit:
Add get-started category

moorepants commented:
We've typically had developer docs in the sympy wiki. Is it the plan that they stay there?

moorepants commented on doc/src/conf.py:
```diff
@@ -30,6 +30,14 @@
+redirects = {
+    "install.rst": "getting_started/install.rst",
+    "aboutus.rst": "getting_started/aboutus.rst",
+    "wiki.rst": "getting_started/wiki.rst",
```
The wiki page is quite brief https://docs.sympy.org/dev/wiki.html and if the wiki houses the development docs, then maybe we only need a link pointing developers to the wiki. This page could be deleted.

moorepants commented on doc/src/conf.py:
```diff
@@ -30,6 +30,14 @@
 
+redirects = {
+    "install.rst": "getting_started/install.rst",
+    "aboutus.rst": "getting_started/aboutus.rst",
```
This has a list of contributors (not sure if this matches the AUTHORS file), the license, and a list of financial support. I think the list of contributors and financial support better on the sympy website, instead of the docs.

moorepants commented on doc/src/conf.py:
```diff
@@ -30,6 +30,14 @@
+    "aboutus.rst": "getting_started/aboutus.rst",
+    "wiki.rst": "getting_started/wiki.rst",
+    "citing.rst": "getting_started/citing.rst",
+    "outreach.rst": "getting_started/outreach.rst",
```
This page is also not so useful. I'd delete and move any useful content to the sympy website.

moorepants commented on doc/src/getting_started/aboutus.rst:
```diff
@@ -4,7 +4,7 @@ About
 ----------------------
 
-.. literalinclude:: ../../AUTHORS
+.. literalinclude:: ../../../AUTHORS
```
I see, this is the actual AUTHORS file. That's good that it is. But I don't see why it belongs in the documentation.

asmeurer replied:
It used to be a curated list of people and what they contributed, but we moved to just copying AUTHORS because curating a list of contributors doesn't scale well. I think it's fine to have a page for it in the docs. That way people who have contributed can get recognition and have somewhere to point to to show that.

moorepants replied:
As per my other comment, I think this would be more appropriate and visible on the website.

asmeurer replied:
I can see what you mean, but it would be lot harder technically to include it on the website. The AUTHORS file lives in the repo so it's easy to include it in the docs build, which gets updated with every release. For the website, we would probably need to include it as part of the release script. I'm not opposed to that, but I think it might be out of scope for @nanjekyejoannah's project.

moorepants replied:
That's fine. There are just things in the sphinx docs that don't really fit in the new four categories and there are things in wiki that probably belong in these new categories. I assumed getting this sorted out was part of the project. Maybe it isn't.

asmeurer replied:
We decided earlier to leave mostly out of scope moving stuff from the wiki into the docs, because that is a lot more work and just cleaning up what is in the docs already is a bit of work. So we should make the categories based on the idea that things will be moved eventually, but unless there is time at the end, we likely won't be able to do it as part of this project. Moving the authors to the website is probably something we should just open an issue about.

moorepants replied:
Sounds good!

moorepants commented on doc/src/index.rst:
```diff
@@ -14,20 +15,73 @@ If you are new to SymPy, start with the :ref:`Tutorial <tutorial>`.
+:ref:`Contributing <contributing>`
+-------------------------------------------------------
+
+Covers instructions for SymPy contributors.
```
If we have this section, will you be moving content from the wiki here? As that is where it is housed now.

nanjekyejoannah replied:
As discussed with @asmeurer and @oscarbenjamin , it's my understanding that we are not moving things from the wiki. In my initial audit, I suggested doing so but I was told it's out of scope for the project.

asmeurer replied:
Yes, we should make room for stuff eventually being moved from the wiki, so that when it happens we have a place for it to go. But moving it is if not out of scope at least the lowest priority thing in this project, compared to cleaning up what is already in the docs.

moorepants replied:
Maybe at least issues could be opened up to encourage people to build out these new categories, particularly with wiki content. It would make for some nice first contributor issues.

nanjekyejoannah commented:
Can I get some consensus on which `rst` files should remain under this `getting started category`. Otherwise, what I gather is that the pages `wiki` and `outreachy` and `aboutus` should be removed from the docs. I plan to update the PR latest tomorrow after getting a response.

asmeurer commented:
The wiki and outreach pages probably can be removed. If anything the wiki page can be put in the developer section, since most of the wiki is developer documentation. The outreach page is perhaps useful, but I don't know if it belongs in "getting started". I think we should keep aboutus, although I don't know what category it should go in. It also doesn't seem to be related to getting started. Maybe there should be another section for this sort of miscellaneous information about the project.

nanjekyejoannah commented:
I updated this first PR to contain the new proposed documentation layout and getting-started section.

For context, I added a SymPy overview section to enhance the general flow.

asmeurer commented:
Just to check, is this in a state where we should be able to merge it now (with additional changes coming in future PRs)? I built the docs on this branch and it seems like it to me, in the sense that every page in the docs can still be accessed from the main docs page, or from direct links, but I want to confirm this before I click merge here.

asmeurer commented on doc/src/index.rst:
```diff
@@ -14,20 +15,67 @@ If you are new to SymPy, start with the :ref:`Tutorial <tutorial>`.
+:ref:`Getting started <installation>`
+-------------------------------------------------------
+
+An introduction to SymPy for developers and detailed tutorials.
```
This is not really accurate. The section as it is now is just installation instructions.

oscarbenjamin commented on doc/src/getting_started/install.rst:
```diff
@@ -1,5 +1,17 @@
+SymPy Overview
+----------------
+
+SymPy is a computer algebra system (CAS) written in the Python programming language. SymPy is easy to use and install (see the download instructions and tutorial for more information), and works everywhere where Python 2.7 or newer is installed (Linux, Windows, Mac OS X, ...). SymPy's features include:
```
This test is outdated. It would be better to take the first two paragraphs (about and sympy is...) from here:
https://www.sympy.org/en/index.html

Also it would be better at the top of the main index page rather than the install page.

nanjekyejoannah replied:
Moved all docs to index, trying to avoid a ToC for a section that is still this small, or you wanted the installation ToC @oscarbenjamin ?

oscar-uob replied:
I don't understand what you are asking

oscarbenjamin replied:
I don't understand what you are asking.

nanjekyejoannah replied:
Creating a separate index page then adding the installation page will lead to a small doc for the category plus a Table of contents for the installation section when you navigate to the getting-started section. Was this intended?

nanjekyejoannah replied:
re: Also it would be better at the top of the main index page rather than the install page.

oscarbenjamin replied:
I meant to move this to the actual main index page. The very first page of the whole docs website (`doc/src/index.rst`).

asmeurer commented on doc/src/conf.py:
```diff
@@ -30,6 +30,10 @@
               'sphinx.ext.mathjax', 'numpydoc', 'sympylive',
               'sphinx.ext.graphviz', 'matplotlib.sphinxext.plot_directive']
 
+redirects = {
```
I think you need to add the extension to make this actually work.

oscarbenjamin commented:
Looks good, thanks.
